### ISSUE TITLE
Generic type with struct constraint creates error on grain interface codegen.

### ISSUE THREAD

**Issue by bstead** [dotnet/orleans#1177]
Its possible I am unknowingly attempting to do something strange here, but I believe I have found an issue with version 1.0.1.

I have a class defined with the constraint as follows
`public class ExampleClass<T>  where T : struct`

My grain interface returns this type as part of a grain:
`Task<ExampleClass<int>> Parse(Message Message);`

After rebuilding the GrainInterface, I end up with the following code in orleans.codegen.cs:
```C#
internal class Common_ExampleClassSerialization\<T>
        where T :  struct, ValueType, new ()
```

The following error presents as a result of this:
```
Error	129	The 'new()' constraint cannot be used with the 'struct' constraint	  
C:\code\Connect\GrainInterface\Properties\orleans.codegen.cs	827	39	GrainInterface
```

 I will continue to dig in a bit, but any assistance you can provide would be much appreciated.

**Comment by bstead** [comment]
I edited this to properly escape the \<>'s

**Comment by gabikliot** [comment]
Fixed via #1178.
